components/private/ShapeLibrary/tests/unit/ShapeLibraryTests.m

```diff
@@ -102,6 +103,40 @@ - (void)testPercentageValueEqualityForCorners {
   XCTAssertEqualObjects(corner, cornerTreatment);
 }
 
+- (void)testPathGeneratorForPercentages {
+  // Given
+  MDCRectangleShapeGenerator *shapeGenerator = [[MDCRectangleShapeGenerator alloc] init];
+  MDCCutCornerTreatment *corner = [[MDCCutCornerTreatment alloc] initWithCut:0.5f];
+  corner.valueType = MDCCornerTreatmentValueTypePercentage;
+  [shapeGenerator setCorners:corner];
+
+  // When
+  CGPathRef path = [shapeGenerator pathForSize:CGSizeMake(100, 100)];
+  NSMutableArray *pathPoints = [NSMutableArray array];
+  CGPathApply(path, (__bridge void *)pathPoints, GetCGPathAddLineToPointValues);
+
+  // Then
+  // The outcome of an 100x100 square with 50% cut corners is a diamond with points at
+  // (0, 50), (50, 0), (100, 50), (50, 100)
+  XCTAssertEqual([pathPoints count], 8);
+  for (NSValue *value in pathPoints) {
+    CGPoint point = value.CGPointValue;
+    XCTAssert(MDCCGFloatEqual(MDCRound(point.x), 0.f) || MDCCGFloatEqual(MDCRound(point.x), 50.f) ||
+              MDCCGFloatEqual(MDCRound(point.x), 100.f));
+    XCTAssert(MDCCGFloatEqual(MDCRound(point.y), 0.f) || MDCCGFloatEqual(MDCRound(point.y), 50.f) ||
+              MDCCGFloatEqual(MDCRound(point.y), 100.f));
+  }
```

Comment:
Suggestion: compare against an array of points.

Reply:
done